### ISSUE TITLE
Request to add allowSingleLineArrow flag for brace-style

### ISSUE THREAD

**Issue by jimjenkins5** [eslint/eslint#11217]
**What rule do you want to change?**

brace-style

**Does this change cause the rule to produce more or fewer warnings?**

Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

I propose that a new option be add to the rule. The option would be `allowSingleLineArrow`, boolean, defaulting to false.

**Please provide some example code that this change will affect:**

```js
functionWithCallback((var1) => { doSomethingWithoutAReturn(); });
```

**What does the rule currently do for this code?**

The following error is thrown:

```
error  Statement inside of curly braces should be on next line             brace-style
```

**What will the rule do after it's changed?**

If `allowSingleLineArrow` is set to true, the preceding code would be allowed. However, the following code would still error:

```
if (a === b) { doSomething(); }
```

**Are you willing to submit a pull request to implement this change?**

Yes

**Comment by jimjenkins5** [comment]
I'd like to note that I understand the above error can be suppressed in 1 of 2 ways, but neither is sufficient.
 1. I could turn on `allowSingleLine`, but that would also allow single line if statements, etc.
 2. I could right the arrow function like: 
`functionWithCallback((var1) => doSomethingWithoutAReturn());`
but then the function is returning the value of `doSomehtingWithoutAReturn()` which may cause problems and will change how the code works.

**Comment by not-an-aardvark** [comment]
Hi, thanks for the proposal.

I'm not sure this option would make sense to add to ESLint core -- it seems like a fairly specific stylistic preference. We wouldn't be able to add individual options for every user's preference, so instead we generally only add an option to a rule if it would be useful for a fairly large percentage of our users, and in other cases we suggest that people [create a custom rule](https://eslint.org/docs/developer-guide/working-with-rules) instead. Personally, I'm not convinced this option would meet the bar for adding to ESLint core, although I wouldn't object to adding it if another team member thinks it's worthwhile.

**Comment by jthomerson** [comment]
@not-an-aardvark I appreciate your clear explanation, but I disagree with your viewpoint that it's overly specific. I think many folks have a brace style for things like conditionals, functions, and other true block elements where they don't want a single line, _but_ they're willing to allow a single line for arrow functions for brevity.

For example, in our standards, we have this:

```js
if (someCondition) { weDoNotWantConditionalBlocksToBeOnSingleLines(); }
becauseWhenYouScanTheCodeYouMightThinkThisIsConditional();
```

On the other hand, we allow it for arrow functions because it's clear when you see an arrow function that you need to look at the function body - to the right. Generally the locations arrow functions are used, you're not likely to misread something in vertical scanning (like you might in the previous example).

We actually did already create a forked version of this rule in [eslint-plugin-silvermine](https://github.com/silvermine/eslint-plugin-silvermine). It was only a few lines of change from the one that's included in core (as you see in the PR #11227). We'd greatly appreciate if you would consider including it in core. Thanks!

**Comment by jimjenkins5** [comment]
@not-an-aardvark Thank you for giving this your attention!

In addition to what @jthomerson brought out, I would also like to mention that ESLint core already has a rule to force block style arrow functions ([arrow-body-style](https://eslint.org/docs/rules/arrow-body-style)). And in the documentation for that rule, numerous examples of single line, block style arrow functions are given.  So allowing them doesn't seem to be a "fairly specific stylistic preference". 

Also, the only way I know of to write a custom rule to accomplish this is to completely fork the core rule and make changes. I don't know of any way to add a rule that conditionally turns off a core rule. If there is a way, please let me know, I would be very interested in that ability.  Forking the core rule opens up problems that we've run into before. Namely: having to keep it up to date with the core version for bug fixes, etc. 

Again, thank you for your consideration. I really appreciate all the hard work you and all the team members put into this project.

**Comment by jthomerson** [comment]
@not-an-aardvark Could someone please reevaluate this request? @jimjenkins5 already submitted a pull request for this, and it's literally a three line code change. But as @jimjenkins5 mentions, the only way for us to implement it outside of eslint is to fork the entire rule (and all of its' tests) into our own plugin. We've done that before when we had to support a feature for indentation, and that's a nasty approach, especially when the suggested change is so minor in nature.

Thanks in advance for giving this a second thought. I'm not familiar enough with the eslint community to know who else may be interested in this, but perhaps you could ping a few others who might have an interest to get their consensus? If we see that several core developers actually don't want to add this feature, it'd be easier to accept than it is when all we see is that (maybe) nobody's even looked at it yet, and then it's killed by a bot.

Thanks again.

**Comment by not-an-aardvark** [comment]
Hi, sorry about the lack of activity here. You might be interested in [`eslint-rule-composer`](https://github.com/not-an-aardvark/eslint-rule-composer) -- it's a tool that would allow you to make some minor changes to rules for your project without forking the whole thing, and I think it would work well for a change like this.
